protos/flyteidl/core/tasks.proto

```diff
@@ -92,6 +92,9 @@ message TaskMetadata {
     oneof interruptible_value {
         bool interruptible = 8;
     };
+
+    // Indicates whether the system should attempt to execute discoverable instances in serial to avoid duplicate work
+    bool discovery_serializable = 9;
```

Comment:
i hate this name. I guess we want to keep it so.
but, ideally it should be cache_serializable right?

Reply:
done. i'm ok with the switch, it hurt me to try and keep it uniform.